scene/resources/default_theme/default_theme.cpp

```diff
@@ -982,7 +982,7 @@ void make_default_theme(bool p_hidpi,Ref<Font> p_font) {
 	Ref<BitmapFont> default_font;
 	if (p_font.is_valid()) {
 		default_font=p_font;
-	} if (p_hidpi) {
+	} else if (p_hidpi) {
```

Comment:
And indeed that's the expected logic: https://github.com/godotengine/godot/blob/de0045cf1b0a5e20fbf74da192039d344ee8d0c7/scene/register_scene_types.cpp#L284-L300

Basically the code that calls `make_default_theme` loads a custom font if one is defined, but does not do the validity check, so it has to happen in `make_default_theme` (and if no font is provided or if it's invalid, use the default fonts).